### PR TITLE
fix: vllm serving with multiple gpus not shutting down properly

### PR DESCRIPTION
**Pull request by leseb**
Internally, we run vLLM using the Python interpreter as described in the [official
documentation](https://docs.vllm.ai/en/latest/serving/openai_compatible_server.html).

vLLM is intended to run via the CLI and be terminated by sending SIGINT (Ctrl+C). This way, when the signal is received, vLLM will properly shut down and reap its child processes.

Another approach was developed by @alimaredia
[here](https://github.com/alimaredia/instructlab/commit/8edfb81ef915b65179bb0f07c94ff82488c8bdbd).
This method involved listing and registering child processes launched by
vLLM, then manually killing each child process during shutdown. @tiran
and I agree that this approach is overkill and should only be used if
there is an urgent bug in vLLM that needs to be fixed quickly. However,
this is not the case, as vLLM correctly responds to SIGINT and
terminates its child processes.

Closes: https://github.com/instructlab/instructlab/issues/1601
Signed-off-by: Sébastien Han <seb@redhat.com>

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by tiran — APPROVED** [review]

**Comment by alimaredia** [comment]
Thanks for looking into this deeper, @leseb and @tiran. I put a hold on the PR because I want one of the vLLM maintainers to take a look at our approach before it goes in. Hopefully that can happen ASAP.

**Comment by alimaredia** [comment]
@leseb FYI  it was @JamesKunstle who originally noticed sending SIGINT instead of SIGTERM to vLLM fixed the issue. 

Could he be credited in the commit message? Feel free to remove me from it, as I just threw together a commit that did exactly what he had put together yesterday.

**Comment by leseb** [comment]
> @leseb FYI it was @JamesKunstle who originally noticed sending SIGINT instead of SIGTERM to vLLM fixed the issue.
> 
> Could he be credited in the commit message? Feel free to remove me from it, as I just threw together a commit that did exactly what he had put together yesterday.

Co-authored you both.

**Review by JamesKunstle — APPROVED** [review]

**Review by danmcp — APPROVED** [review]

File: src/instructlab/model/backends/backends.py [hunk 1]
```diff
@@ -201,19 +201,21 @@ def get(model_path: pathlib.Path, backend: str | None) -> str:
     return backend
 
 
+# TODO: This is only used by vLLM but should move to vllm.py
```
Comment (njhill) [inline src/instructlab/model/backends/backends.py:204]: Any reason not to do that in this PR too?

**Review by njhill — APPROVED** [review]
LGTM ... we'll also aim to make an upstream change to treat `SIGTERM` similarly.

**Review by alimaredia — APPROVED** [review]

File: src/instructlab/model/backends/backends.py (hunk 1 above)
Comment (leseb) [reply src/instructlab/model/backends/backends.py:204]: Because they are code refactors happening on the side of this PR that will pick it up :)

**Pull request merged**